nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/convolution/Upsampling3d.java

```diff
@@ -0,0 +1,99 @@
+/* ******************************************************************************
+ * Copyright (c) 2020 Konduit K.K.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+package org.nd4j.linalg.api.ops.impl.layers.convolution;
+
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.nd4j.autodiff.samediff.SDVariable;
+import org.nd4j.autodiff.samediff.SameDiff;
+import org.nd4j.base.Preconditions;
+import org.nd4j.linalg.api.buffer.DataType;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.api.ops.DynamicCustomOp;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Upsampling3d operation
+ */
+@Slf4j
+@Getter
+@NoArgsConstructor
+public class Upsampling3d extends DynamicCustomOp {
+
+
+    protected boolean NCDHW;
+    protected int scaleH;
+    protected int scaleW;
+    protected int scaleD;
+
+    public Upsampling3d(SameDiff sameDiff, SDVariable input, boolean NCDHW, int scaleD, int scaleH, int scaleW) {
+        super("upsampling3d",sameDiff, new SDVariable[]{input});
+        this.NCDHW = NCDHW;
```

Comment:
The convention in java is that non-final fields are lowercase or camelCase
So `ncdhw` -> `ncdhw`

all caps should only be used for constants (with final modifier) like:
`public static final double PI = 3.14159`